### PR TITLE
829 - YouTube videos don't set the a thumbnail 

### LINKED ISSUE #829: YouTube videos don't set the a thumbnail

### PR DESCRIPTION
**Pull request by yungalyx**
Closes #829

### CODE REVIEW

**Comment by seanRong** [comment]
Works good for me,

Noticed that my .env file got changed somewhere along the line and I lost my Link Preview API key and that caused the thumbnails to not load. Wonder if that should be in gitignore?

**Comment by narendrasss** [comment]
Is this ready for review @yungalyx? If so, can you set the PR as ready for review? (i.e. not a draft)

**Comment by yungalyx** [comment]
@narendrasss yess! I thought the reviewers are allowed to set status to "ready for review" as well

**Comment by narendrasss** [comment]
@yungalyx I am yes! Just making sure 😄

File: templates/vue/src/components/NodeModal.vue [hunk 1]
```diff
@@ -475,7 +475,10 @@ export default {
         this.updateNodeCoordinates()
         this.loading = true
 
-        if (this.node.mediaType === "url-embed" && this.node.behaviour !== "embed") {
+        if (
+          (this.node.mediaType === "url-embed" && this.node.behaviour !== "embed") ||
+          this.node.mediaFormat === "youtube"
+        ) {
```
Comment (narendrasss) [inline templates/vue/src/components/NodeModal.vue:481]: Can you change this to a computed property? Would be more readable given that it's used in multiple places.

**Review by narendrasss — CHANGES_REQUESTED** [review]
Small nitpick:

File: templates/vue/src/components/NodeModal.vue (hunk 1 above)
Comment (yungalyx) [reply templates/vue/src/components/NodeModal.vue:481]: Done!

**Comment by wynnset** [comment]
Hey @yungalyx this PR is assigned to you, but marked as ready for review. Is that an oversight or are you still doing something with this PR?

**Comment by yungalyx** [comment]
@wynnset my bad! forgot to unassign myself.

**Review by narendrasss — APPROVED** [review]

**Review by wynnset — APPROVED** [review]

**Comment by wynnset** [comment]
Sorry @yungalyx it's not clear to me what this PR actually does. When I tested this, YouTube still does not add a thumbnail. Perhaps the title of this PR is incorrect - in that case, could you please fix it?

**Comment by yungalyx** [comment]
@wynnset This PR automatically adds a thumbnail for YouTube video nodes...  Can you send me the url that isnt working for you?

**Comment by wynnset** [comment]
@yungalyx I do get the confirmation box asking me if I want to use the preview link image as the thumbnail, so that part works fine. But when I do say I want that, there are 2 issues:
1. While the images are being retrieved, currently nothing shows that anything is happening. I just have to wait until it finishes. In the meantime, I can press "Publish" again or press Cancel. We should put a loading screen on while the thumbnail is being updated. 
2. The thumbnail gets saved, but it does not show up on the node for some reason. When I inspect the code, I see it's there, but it just doesn't show up.

**Comment by yungalyx** [comment]
@wynnset Issue 1 is being addressed in PR #850 and I cannot recreate the 2nd issue on my end.. I'm able to get a rendered thumbnail for all youtube videos (aside from privated videos). I'm going to currently block this PR as I work on #850 and if you can drop me the link that isn't working for you, I can give this another shot.

**Comment by wynnset** [comment]
Ok. I just tested it on another tapestry and it worked so it might be a data issue on my end. That sounds good!

**Pull request merged**